solidity/ecdsa/test/WalletRegistry.Authorization.test.ts

```diff
@@ -0,0 +1,2934 @@
+    // Staking provider address is unique for each stake delegation - see T
+    // TokenStaking contract.
+    context(
+      "when operator has been already registered for the staking provider",
+      () => {
+        before(async () => {
+          await createSnapshot()
+          await walletRegistry
+            .connect(stakingProvider)
+            .registerOperator(operator.address)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            walletRegistry
+              .connect(stakingProvider)
+              .registerOperator(operator.address)
+          ).to.be.revertedWith("Operator already set for the staking provider")
+
+          // should revert even if it's another operator than the one previously
+          // registered for the staking provider
+          await expect(
+            walletRegistry
+              .connect(stakingProvider)
+              .registerOperator(thirdParty.address)
+          ).to.be.revertedWith("Operator already set for the staking provider")
+        })
+      }
+    )
+
+    // Some other staking provider is using the given operator address.
+    // Should not happen in practice but we should protect against it.
+    context("when the operator is already in use", () => {
+      before(async () => {
+        await createSnapshot()
+        await walletRegistry
+          .connect(thirdParty)
+          .registerOperator(operator.address)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should revert", async () => {
+        await expect(
+          walletRegistry
+            .connect(stakingProvider)
+            .registerOperator(operator.address)
+        ).to.be.revertedWith("Operator address already in use")
+      })
+    })
+
+    // This is the normal, happy path. Stake owner delegated their stake to
+    // the staking provider, and the staking provider is registering operator
+    // for ECDSA application.
+    context("when staking provider is registering new operator", () => {
+      let tx: ContractTransaction
+
+      before(async () => {
+        await createSnapshot()
+        tx = await walletRegistry
+          .connect(stakingProvider)
+          .registerOperator(operator.address)
+      })
+
+      after(async () => {
+        await restoreSnapshot()
+      })
+
+      it("should set staking provider -> operator mapping", async () => {
+        expect(
+          await walletRegistry.stakingProviderToOperator(
+            stakingProvider.address
+          )
+        ).to.equal(operator.address)
+      })
+
+      it("should set operator -> staking provider mapping", async () => {
+        expect(
+          await walletRegistry.operatorToStakingProvider(operator.address)
+        ).to.equal(stakingProvider.address)
+      })
+
+      it("should emit OperatorRegistered event", async () => {
```

Comment:
We could also add a check that the operator wasn't registered in the sortition pool.

Reply:
https://github.com/keep-network/keep-core/pull/2881